### ISSUE TITLE
Handling of errSSLPeerAuthCompleted and other non-fatal error codes

### ISSUE THREAD

**Issue by 0xADD1E** [sfackler/rust-native-tls#205]
When attempting to do peer authentication with a macOS client, an error gets raised corresponding with errSSLPeerAuthCompleted. This shouldn't actually be raised, should it?

**Comment by sfackler** [comment]
Yeah, you should not be seeing an errSSLPeerAuthCompleted. Could you put together a small self contained example that hits that error?

**Comment by ragoso** [comment]
Hi, I'm getting the same "peer cert is valid, or was ignored if verification disabled" error from using reqwest with the "native-tls" feature on a MacOS M1.

My implementation loads a PKCS12 as an identity to send in the request.

Investigating, I found this line in the reqwest project that indicates the use of tokio_native_tls.

https://github.com/seanmonstar/reqwest/blob/7e7b116a134cc0d6d646ab316dd83976369d5298/src/connect.rs#LL245C37-L245C53

It seems to be something with this comment on Apple's SecureTransport.h.

```
  * errSSLPeerAuthCompleted: Peer's cert chain is valid, or was ignored if
  * cert verification was disabled via SSLSetEnableCertVerify. The application
  * may decide to continue with the handshake (by calling SSLHandshake
  * again), or close the connection at this point.
```

**Comment by ragoso** [comment]
Hey guys,

[this PR](https://github.com/kornelski/rust-security-framework/pull/183) at security-framework solves this problem.
Release 2.9.1
